### PR TITLE
Remove double-iteration from the mime wall code

### LINKED ISSUE #16711: Mime wall issue
https://github.com/space-wizards/space-station-14/pull/16701 is double-iterating and using an obsolete method and needs fixing.

### PR DESCRIPTION
**Pull request by Winkarst-cpu**
## About the PR
Removed double-iteration from the mime wall code.

## Why / Balance
Resolves #16711 (but I don't understand what method is obsolete).

## Technical details
Not much to say...
изображение

## Media

https://github.com/user-attachments/assets/efaee2e1-edea-4c71-b101-85d69d22e110

## Requirements

- [X] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

None.

### CODE REVIEW

**Comment by TheShuEd** [comment]
you removed the mobs check, you shouldn't do that. Otherwise the mime will be able to make walls right inside people

**Comment by TheShuEd** [comment]
the problem is that the code checks all entities around it 2 times in a row - first for walls, then for creatures. It is necessary to optimize the checks in one iteration of the loop.

**Review by TheShuEd — CHANGES_REQUESTED** [review]
!

**Comment by Winkarst-cpu** [comment]
> you removed the mobs check, you shouldn't do that. Otherwise the mime will be able to make walls right inside people

Uh... ``CollisionGroup.AllMask`` includes mobs.

https://github.com/user-attachments/assets/1a3d4187-8893-499e-ba20-8b998cdb72cf

**Comment by Winkarst-cpu** [comment]
Bro speaking in symbols :trollface:

File: Content.Server/Abilities/Mime/MimePowersSystem.cs [hunk 1]
```diff
@@ -80,22 +78,13 @@ private void OnInvisibleWall(EntityUid uid, MimePowersComponent component, Invis
             if (tile == null)
                 return;
 
-            // Check there are no walls there
-            if (_turf.IsTileBlocked(tile.Value, CollisionGroup.Impassable))
+            // Check if the tile is blocked by a wall or mob, and don't create the wall if so
+            if (_turf.IsTileBlocked(tile.Value, CollisionGroup.MobLayer | CollisionGroup.WallLayer))
```
Comment (slarticodefast) [inline Content.Server/Abilities/Mime/MimePowersSystem.cs:82]:
```suggestion
            if (_turf.IsTileBlocked(tile.Value, CollisionGroup.Impassable | CollisionGroup.Opaque))
```
Ok, I did some more testing and I think it should be this instead. The Impassable layer is walls and such (we had that part before) and the Opaque layer is making sure we cannot put it over mobs (Opaque is part of MobLayer), but we can still place it over tables and machines like before.

**Review by slarticodefast — CHANGES_REQUESTED** [review]

File: Content.Server/Abilities/Mime/MimePowersSystem.cs (hunk 1 above)
Comment (Winkarst-cpu) [reply Content.Server/Abilities/Mime/MimePowersSystem.cs]: Done.

**Review by slarticodefast — APPROVED** [review]

**Comment by slarticodefast** [comment]
Thanks for the cleanup!

**Pull request merged**